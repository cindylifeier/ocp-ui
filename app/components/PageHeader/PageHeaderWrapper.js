import styled from 'styled-components';

const PageHeaderWrapper = styled.div`
  padding-left: 1vw;
  padding-top: 1vh;
  padding-bottom: 1vh;
  font-size: 1.5rem;
  font-weight: bold;
  color: #366;
`;

PageHeaderWrapper.propTypes = {};

export default PageHeaderWrapper;
