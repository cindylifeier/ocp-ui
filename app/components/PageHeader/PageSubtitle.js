import styled from 'styled-components';

const PageSubtitle = styled.div`
  padding-left: 0.5vw;
  margin-top: 2vh;
  margin-left: 1vw;
  margin-right: 1vw;
  font-size: 1.2rem;
  font-weight: bold;
  color: #444;
  background-color: #f2f2f2;
`;

PageSubtitle.propTypes = {};

export default PageSubtitle;
