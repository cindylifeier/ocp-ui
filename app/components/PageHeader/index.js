/**
 *
 * PageHeader
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Divider } from 'material-ui';

import PageHeaderWrapper from './PageHeaderWrapper';

function PageHeader({ children }) {
  return (
    <div>
      <PageHeaderWrapper>{children}</PageHeaderWrapper>
      <Divider />
    </div>
  );
}

PageHeader.propTypes = {
  children: PropTypes.node,
};

export default PageHeader;
