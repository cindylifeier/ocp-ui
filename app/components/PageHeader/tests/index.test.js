import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';
import { Divider } from 'material-ui';

import PageHeader from '../index';
import PageHeaderWrapper from '../PageHeaderWrapper';

configure({ adapter: new Adapter() });

describe('<PageHeader />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<PageHeader>{children}</PageHeader>);

      // Arrange
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('structural tests', () => {
    it('should contain children', () => {
      // Arrange
      const children = <div>test</div>;

      // Act
      const renderedComponent = shallow(<PageHeader>{children}</PageHeader>);

      // Arrange
      expect(renderedComponent.contains(children)).toEqual(true);
    });

    it('should contain <PageHeaderWrapper>{children}</PageHeaderWrapper>', () => {
      // Arrange
      const children = <div>test</div>;
      const expected = (<PageHeaderWrapper>{children}</PageHeaderWrapper>);

      // Act
      const renderedComponent = shallow(<PageHeader>{children}</PageHeader>);

      // Arrange
      expect(renderedComponent.contains(expected)).toEqual(true);
    });

    it('should contain <Divider />', () => {
      // Arrange
      const children = <div>test</div>;
      const expected = (<Divider />);

      // Act
      const renderedComponent = shallow(<PageHeader>{children}</PageHeader>);

      // Arrange
      expect(renderedComponent.contains(expected)).toEqual(true);
    });
  });
});
