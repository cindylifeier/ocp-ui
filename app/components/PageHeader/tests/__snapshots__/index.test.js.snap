// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageHeader>
    <div>
      test
    </div>
  </PageHeader>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PageHeaderWrapper__PageHeaderWrapper>
          <div>
            test
          </div>
        </PageHeaderWrapper__PageHeaderWrapper>,
        <Divider
          inset={false}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            test
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "inset": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PageHeaderWrapper__PageHeaderWrapper>
            <div>
              test
            </div>
          </PageHeaderWrapper__PageHeaderWrapper>,
          <Divider
            inset={false}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              test
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test",
            },
            "ref": null,
            "rendered": "test",
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "inset": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
